apps/src/dance/ai/utils/getItem.ts

```diff
@@ -0,0 +1,11 @@
+import {DanceAiModelItem} from '../types';
+
+import inputLibraryJson from '@cdo/static/dance/ai/ai-inputs.json';
+
+const itemsById: {[key: string]: DanceAiModelItem} =
+  inputLibraryJson.items.reduce(
+    (bucket, item) => ({...bucket, [item.id]: item}),
+    {}
+  );
+
```

Comment:
the original function would search the array every time (and was also redefined on every render). Instead, build a hash table of item id -> item, and then just key into that for our search function. Substantial speed boost!